test/utils/interface-common-factory.js

```diff
@@ -68,25 +73,23 @@ function createFactory (options) {
   }
 }
 
-function createAsync (createFactoryOptions = {}, createSpawnOptions = {}) {
+function createAsync (options = {}) {
   return () => {
     const nodes = []
-    const setup = async (factoryOptions = {}, spawnOptions = {}) => {
-      factoryOptions = mergeOptions(
-        {
-          type: 'proc',
-          exec: IPFS
-        },
-        factoryOptions,
-        createFactoryOptions
+    const setup = async (setupOptions = {}) => {
+      options.factoryOptions = mergeOptions(
+        setupOptions.factoryOptions,
+        options.factoryOptions || DEFAULT_FACTORY_OPTIONS
       )
```

Comment:
The order here has changed?

Reply:
No. `factoryOptions` is now `setupOptions.factoryOptions` and `createFactoryOptions` is now `options.factoryOptions`.
So the order is still the same!

Reply:
Right but the defaults are now last. Should this be:

```js
      options.factoryOptions = mergeOptions(
        { ...DEFAULT_FACTORY_OPTIONS },
        setupOptions.factoryOptions,
        options.factoryOptions
      )
```

Reply:
Got it. The thing here is: we only want to use `DEFAULT_FACTORY_OPTIONS` if there's no `options.factoryOptions`.

Should we do something like:
```js
options.factoryOptions = mergeOptions(
     options.factoryOptions ? { ...DEFAULT_FACTORY_OPTIONS } : {},
     setupOptions.factoryOptions,
     options.factoryOptions
)
```

Reply:
yeah that works 👍